File: ICAV Time Tracker/ICAV Time Tracker/JobAssignment.swift
//
//  JobAssignment.swift
//  ICAV Time Tracker
//
//  Created by Jason Baker on 6/23/25.
//

import Foundation

struct JobAssignment: Codable, Identifiable {
    let id: String
    let jobId: String
    let userId: String
    let technicianName: String
    let assignedDate: Date
    let assignedHours: Int
    let actualHours: Int?
    let status: String
    let notes: String?
    let order: Int?
    let createdAt: Date
    let updatedAt: Date
    let job: Job?
}

struct Job: Codable, Identifiable {
    let id: String
    let title: String
    let customerName: String
    let description: String?
    let location: String?
    let estimatedHours: Int
    let status: String
    let priority: String
    let createdAt: Date
    let updatedAt: Date
} 

File: Android App/app/src/main/java/com/example/icavtimetracker/data/JobAssignment.kt
package com.example.icavtimetracker.data

import java.util.Date

data class JobAssignment(
    val id: String,
    val jobId: String,
    val userId: String,
    val technicianName: String,
    val assignedDate: Date,
    val assignedHours: Int,
    val actualHours: Int? = null,
    val status: String = "assigned",
    val notes: String? = null,
    val order: Int? = null,
    val createdAt: Date,
    val updatedAt: Date,
    val job: Job? = null
)

data class Job(
    val id: String,
    val title: String,
    val customerName: String,
    val description: String? = null,
    val location: String? = null,
    val estimatedHours: Int,
    val status: String,
    val priority: String,
    val createdAt: Date,
    val updatedAt: Date
) 